### PR TITLE
Add PPP and use cases, glossary to Developer's Guide

### PR DESCRIPTION
**Pull request by geoffong11**
Glossary may need to be further updated when more information is added to Developer's Guide

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -321,7 +422,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Private contact detail**:  contact detail that is not meant to be shared with others
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:425]: What about making the formatting tidier, like everything starting with uppercase letters?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -321,7 +422,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* **tag**: a fixed label that is meant for easier searching and organisation
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:426]:
```suggestion
* **Tag**: A fixed label that is meant for easier searching and organisation
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -321,7 +422,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+*  **index**: Number that references the position of person in the list
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:427]:
```suggestion
*  **Index**: A number that references the position of person in the list
```

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (imkwokyong) [inline docs/DeveloperGuide.md:425]:
```suggestion
* **Private contact detail**:  A contact detail that is not meant to be shared with others
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -283,32 +283,133 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 **MSS**
 
 1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+2.  Connexion shows a list of persons
+3.  User requests to delete a specific person
+4.  Connexion requests for index of person
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:295]: Do we have this step in our implementation?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -283,32 +283,133 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes at step 4.
+
+* 7a. Connexion detects an error in the entered details
+
+    * 7a1. Connexion shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**Use case: Tag a person for organisation**
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:376]: Do you mean marking? Tagging is implemented in `add` or `update`.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -283,32 +283,133 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 7a. The given tag is invalid.
+    * 7a1. Connexion shows an error message.
+
+      Use case resumes at step 6.
+
+      Use case ends.
+* 7b. The given tag is already present.
+    * 7b1. Connexion tells the user the tag is already present.
+
+      Use case ends.
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:413]: It would be great to add a use case on `filter` here.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -283,32 +283,133 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
```
Comment (imkwokyong) [inline docs/DeveloperGuide.md:284]: Will it be better to keep the use cases simpler first?

**Review by imkwokyong — COMMENTED** [review]
PPP is good!
However, there are a few suggestions from me on the Developer Guide.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (geoffong11) [reply docs/DeveloperGuide.md:412]: Thanks for the input!

**Pull request closed without merging**